examples/cookbook/app/network-requests/__tests__/test-utils.ts

```diff
@@ -0,0 +1,109 @@
+import { User } from '../types';
+import {http, HttpResponse} from "msw";
+import {setupServer} from "msw/node";
+
+// Define request handlers and response resolvers for random user API.
+// By default, we always return the happy path response.
+const handlers = [
+  http.get('https://randomuser.me/api/*', () => {
+    return HttpResponse.json(DATA);
+  }),
+];
+
+export const server = setupServer(...handlers);
+
+export const mockServerFailureForGetAllContacts = () => {
+  server.use(
+    http.get('https://randomuser.me/api/', ({ request }) => {
+      // Construct a URL instance out of the intercepted request.
+      const url = new URL(request.url);
+      // Read the "results" URL query parameter using the "URLSearchParams" API.
+      const resultsLength = url.searchParams.get('results');
+      // Simulate a server error for the get all contacts request.
+      // We check if the "results" query parameter is set to "25"
+      // to know it's the correct request to mock, in our case get all contacts.
+      if (resultsLength === '25') {
```

Comment:
ok, now i get the idea, so this is because `https://randomuser.me/api` serves only a single endpoint, so you need to differentiate between different calls using the results param. Perhaps we can use `?seed=favorites` to simulate 2 different endpoints. Alternatively use some other test API that has 2 diff endpoints, e.g. swapi.dev. Wdyt?

Reply:
Good idea, I have made it more explicit!